sbt-metals/src/main/scala/scala/meta/sbt/MetalsPlugin.scala

```diff
@@ -39,6 +39,7 @@ object MetalsPlugin extends AutoPlugin {
         // without config scope it will aggregate over all project dependencies
         // and their configurations
         metalsWriteBuildInfo := Def.taskDyn {
+          IO.delete(Metals.buildinfoDir.value / thisProject.value.id)
```

Comment:
Is there a reason we don't remove `.metals/buildinfo` in `metalsSetup`? This will not catch stale/removed projects which means there will be leftover outdated `buildinfo/**.properties` files

Reply:
See https://github.com/scalameta/metals/pull/269#discussion_r179955398. This is taken care of in the definition of `*/metalsWriteBuildInfo`. So when you change your projects structure and call `*/metalsWriteBuildInfo` (or `metalsSetup`), it will delete the whole `.metals/buildinfo/` and write buildinfo for all projects.

Here I remove only project-related buildinfo, because it's a definition of a project-scoped `metalsWriteBuildInfo` which shouldn't touch other projects' stuff.